atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -223,6 +169,71 @@ def _send_mail(self, receivers_list, subject, body):
             if s is not None:
                 s.quit()
 
+    def _get_email_from_koji_obj(self, obj):
+        if obj.get('krb_principal') and obj['krb_principal'] != '':
+            return obj['krb_principal'].lower()
+        else:
+            if not self.email_domain:
+                raise RuntimeError("Empty email_domain specified")
+            return '@'.join([obj['owner_name'], self.email_domain])
+
+    def _get_koji_submitter(self, session, metadata):
+        koji_task_id = metadata['labels']['koji-task-id']
+        koji_task_info = session.getTaskInfo(koji_task_id)
+        koji_task_owner = session.getUser(koji_task_info['owner'])
+        koji_task_owner_email = self._get_email_from_koji_obj(koji_task_owner)
+        self.submitter = koji_task_owner_email
+        return koji_task_owner_email
+
+    def _get_koji_owners(self, session, metadata):
+        result = []
+        koji_build_id = metadata['labels']['koji-build-id']
+        koji_build_info = session.getBuild(koji_build_id)
+
+        koji_tags = session.listTags(koji_build_id)
+        for koji_tag in koji_tags:
+            koji_tag_id = koji_tag['id']
+            koji_package_id = koji_build_info['package_id']
+            koji_pkg_tag_config = session.getPackageConfig(koji_tag_id, koji_package_id)
+            koji_pkg_tag_owner = session.getUser(koji_pkg_tag_config['owner'])
+
+            result.append(self._get_email_from_koji_obj(koji_pkg_tag_owner))
+
+        return result
+
+    def _get_receivers_list(self):
+        receivers_list = []
+        if self.always_notify:
+            receivers_list.append(self.always_notify)
+
+        if self.to_koji_submitter or self.to_koji_pkgowner:
+            try:
+                metadata = get_build_json().get("metadata", {})
```

Comment:
Just to clarify - the plugin can work without Openshift - but the logs would say 'None' there.

Koji integration would require openshift (even `koji_promote` would fail without openshift, as there would be no place to store metadata).